packages/pwa-kit-react-sdk/src/ssr/universal/components/with-react-query/index.js

```diff
@@ -13,16 +13,27 @@ import ssrPrepass from 'react-ssr-prepass'
 const isServerSide = typeof window === 'undefined'
 const STATE_KEY = '__reactQuery'
 
-export const withReactQuery = (Wrapped) => {
+export const withReactQuery = (Wrapped, options = {}) => {
     /* istanbul ignore next */
     const wrappedComponentName = Wrapped.displayName || Wrapped.name
+    const queryClientConfig = options.queryClientConfig || {
```

Comment:
Thinking some more about what Adam wrote last time:
> we also need to think about the potential global configuration combinations a user can define in the user space, that will affect the server-side and client-side queries made by commerce-sdk-react.

One **key insight** is that the pwa sdk needs to be aware of every query that would get run on the server side. This insight leads to our current implementation of reusing the same query client/provider on all the packages (pwa sdk, our commerce hooks, and the user's project).

However, as a consequence, like what Adam said, when the users override the client config, it will then affect all of the packages mentioned above. But it's likely/possible for each package to want to define the config in its own way. For example, currently we know that:
- The pwa sdk wants to do retries only on the client side
- while our commerce hooks are planning to do retries on both server and client sides, but only for http 5xx errors (see PR #718)
- furthermore, the user's project would likely have its own preference too.

So I think it's better for each package to have its own query client/provider.

Reply:
After having a discussion in person, we acknowledged that there could possibly be a use case for having multiple query providers. We currently don't prevent end users from having multiple providers if they wish to have different configurations or simply have separate their queries. But it comes as the cost of not supporting server-side rendering as the SDK currently only knows about the one provider that it creates. 

We talked about how we might support multiple providers in SSR, but at this point in time, we decided that the errors vs the number of use cases for multiple providers doesn't warrant the work.

We also talked about error retries and all aligned on the position that we shouldn't be doing retries on the server. We talked about leaving that problem in user-space. In this PR I warn if they are using retries on the server. But leave the configuration in the template.